packages/playwright-core/src/server/screenshotter.ts

```diff
@@ -260,22 +259,19 @@ export class Screenshotter {
     const framesToParsedSelectors: MultiMap<Frame, ParsedSelector> = new MultiMap();
 
     const cleanup = async () => {
-      await Promise.all([...framesToParsedSelectors.keys()].map(async frame => {
-        await frame.hideHighlight();
+      await Promise.all(this._page.frames().map(async frame => {
+        await frame.unmaskSelectors();
       }));
     };
 
-    if (!options.mask || !options.mask.length)
-      return cleanup;
-
     await Promise.all((options.mask || []).map(async ({ frame, selector }) => {
       const pair = await frame.selectors.resolveFrameForSelector(selector);
       if (pair)
         framesToParsedSelectors.set(pair.frame, pair.info.parsed);
     }));
     progress.throwIfAborted(); // Avoid extra work.
 
-    await Promise.all([...framesToParsedSelectors.keys()].map(async frame => {
+    await Promise.all(this._page.frames().map(async frame => {
```

Comment:
I wonder whether this would never resolve because one ad iframe is perpetually navigating? Should we use `nonStallingEvaluateInExistingContext` for frames that are not explicitly specified by any `mask` locator?

Reply:
If that is an ad frame, we should not go into it at all. Masking in-frame locators sounds artificial to me altogether, it seems like letting the user "annotate" elements with the attributes beforehand is a better approach. With that in mind, having them as separate code paths make sense.